src/main/java/com/openshift/internal/restclient/model/volume/VolumeSource.java

```diff
@@ -14,22 +14,22 @@
 
 import java.util.HashMap;
 
+import com.openshift.restclient.model.volume.VolumeType;
 import org.jboss.dmr.ModelNode;
 
 import com.openshift.internal.restclient.model.ModelNodeAdapter;
 import com.openshift.internal.restclient.model.properties.ResourcePropertyKeys;
-import com.openshift.restclient.model.volume.IVolume;
 import com.openshift.restclient.model.volume.IVolumeSource;
 
 /**
  * 
  * @author Jeff Cantrill
  *
  */
-public class VolumeSource 
+public abstract class VolumeSource
```

Comment:
we should consider modifying this class to extend from KubernetesResource instead of the ModelNodeAdapter since it looks like we can do REST operations on persistentvolumeclaims